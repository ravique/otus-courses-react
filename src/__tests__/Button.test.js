import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import '../../jest.config'

import Button from '../components/button/index'

describe('Button', () => {
    it('should render button', () => {
        const output = shallow(
            <Button url='google.com' text="Show more"/>
        );

        expect(shallowToJson(output)).toMatchSnapshot();
        expect(output.hasClass('button')).toEqual(true);
    });

    it('should click', () => {
        const output = shallow(
            <Button url='google.com' text="Show more"/>
        );

        expect(output.state().pressed).toEqual(false);
        expect(output.hasClass('button_pressed')).toEqual(false);

        output.simulate('click');

        expect(output.state().pressed).toEqual(true);
        expect(output.hasClass('button_pressed')).toEqual(true);

    });
});