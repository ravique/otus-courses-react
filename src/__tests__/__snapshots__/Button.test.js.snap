// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button 1`] = `
<Link
  className="button"
  onClick={[Function]}
  to="google.com"
>
  Show more
</Link>
`;
